apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js

```diff
@@ -0,0 +1,86 @@
+import {
+  MICROBIT_FIRMATA_V1_URL,
+  MICROBIT_FIRMATA_V2_URL,
+  MICROBIT_IDS_V1,
+  MICROBIT_IDS_V2,
+  MICROBIT_V1,
+  MICROBIT_V2
+} from '@cdo/apps/lib/kits/maker/boards/microBit/MicroBitConstants';
+import {DAPLink, WebUSB} from 'dapjs';
+import {getStore} from '@cdo/apps/redux';
+import {setMicrobitFirmataUpdatePercent} from '@cdo/apps/lib/kits/maker/microbitRedux';
+import {EventEmitter} from 'events'; // provided by webpack's node-libs-browser
+
+export default class MBFirmataUpdater extends EventEmitter {
+  constructor() {
+    super();
+    // State value 'microbitFirmataUpdatePercent' is only updated when this instance variable changes.
+    this.firmataUpdatePercent = 0;
+  }
+
+  detectMicrobitVersion(device) {
+    // Detect micro:bit version and select the right Intel Hex for micro:bit V1 or V2
+    const microbitId = device.serialNumber.substring(0, 4);
+    let microbitVersion = MICROBIT_V1;
+    if (MICROBIT_IDS_V2.includes(microbitId)) {
+      microbitVersion = MICROBIT_V2;
+    } else if (!MICROBIT_IDS_V1.includes(microbitId)) {
+      microbitVersion = null;
+    }
+    return microbitVersion;
+  }
+
+  async updateMBFirmataVersioned() {
+    const device = await navigator.usb.requestDevice({
+      filters: [{vendorId: 0x0d28, productId: 0x0204}]
+    });
+    const microbitVersion = this.detectMicrobitVersion(device);
```

Comment:
It looks like this is the only place `detectMicrobitVersion` is used, and then we immediately convert it to the `firmataUrl` using similar logic. Could we name the function `detectFirmataUrlForMicrobitVersion` or just `detectFirmataUrl` and have it return the `firmataUrl` directly?

Reply:
I would like to keep `detectMicroBitVersion` (note that upper case B!) as a separate function in case we want to keep track data of the micro:bit version type our users use in the future. Also, there is a way to update a micro:bit with a universal .hex file which we don't need at this point, but may need later. Here's [the implementation](https://github.com/code-dot-org/code-dot-org/blob/ead90fd947e9d159384d43faf2f9fd98c522338f/apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js#L393).

Reply:
I updated file - kept `detectMicroBitVersion` but added `getFirmatURLByVersion`.